### ISSUE TITLE
Crash when crushing Grapes

### ISSUE THREAD

**Issue by yorkmouse** [cScotPlay/Rusticated#9]
---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 2023-01-31 10:47:40
Description: Rendering Block Entity

java.lang.ExceptionInInitializerError
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.render(CrushingTubRenderer.java:40)
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.method_3569(CrushingTubRenderer.java:23)
	at net.minecraft.class_824.method_23079(class_824.java:90)
	at net.minecraft.class_824.method_23081(class_824.java:79)
	at net.minecraft.class_824.method_3554(class_824.java:104)
	at net.minecraft.class_824.method_3555(class_824.java:79)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderBlockEntities(ShadowRenderer.java:551)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderShadows(ShadowRenderer.java:473)
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.renderShadows(NewWorldRenderingPipeline.java:761)
	at net.minecraft.class_761.handler$znc000$iris$renderTerrainShadows(class_761.java:7034)
	at net.minecraft.class_761.method_22710(class_761.java:1240)
	at net.minecraft.class_757.method_3188(class_757.java:1039)
	at net.minecraft.class_757.method_3192(class_757.java:816)
	at net.minecraft.class_310.method_1523(class_310.java:1177)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NullPointerException: Cannot invoke "net.fabricmc.fabric.api.renderer.v1.Renderer.materialFinder()" because "r" is null
	at net.mcs3.rusticated.util.RenderUtility.<clinit>(RenderUtility.java:49)
	... 20 more

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.render(CrushingTubRenderer.java:40)
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.method_3569(CrushingTubRenderer.java:23)
	at net.minecraft.class_824.method_23079(class_824.java:90)
	at net.minecraft.class_824.method_23081(class_824.java:79)
	at net.minecraft.class_824.method_3554(class_824.java:104)
	at net.minecraft.class_824.method_3555(class_824.java:79)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderBlockEntities(ShadowRenderer.java:551)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderShadows(ShadowRenderer.java:473)
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.renderShadows(NewWorldRenderingPipeline.java:761)
	at net.minecraft.class_761.handler$znc000$iris$renderTerrainShadows(class_761.java:7034)
	at net.minecraft.class_761.method_22710(class_761.java:1240)
	at net.minecraft.class_757.method_3188(class_757.java:1039)

-- Block Entity Details --
Details:
	Name: rusticated:crushing_tub_container // net.mcs3.rusticated.world.level.block.entity.CrushingTubBlockEntity
	Block: Block{rusticated:crushing_tub}
	Block location: World: (-103,64,32), Section: (at 9,0,0 in -7,4,2; chunk contains blocks -112,-64,32 to -97,319,47), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Block: Block{rusticated:crushing_tub}
	Block location: World: (-103,64,32), Section: (at 9,0,0 in -7,4,2; chunk contains blocks -112,-64,32 to -97,319,47), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
Stacktrace:
	at net.minecraft.class_824.method_3554(class_824.java:104)
	at net.minecraft.class_824.method_3555(class_824.java:79)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderBlockEntities(ShadowRenderer.java:551)
	at net.coderbot.iris.pipeline.ShadowRenderer.renderShadows(ShadowRenderer.java:473)
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.renderShadows(NewWorldRenderingPipeline.java:761)
	at net.minecraft.class_761.handler$znc000$iris$renderTerrainShadows(class_761.java:7034)
	at net.minecraft.class_761.method_22710(class_761.java:1240)
	at net.minecraft.class_757.method_3188(class_757.java:1039)
	at net.minecraft.class_757.method_3192(class_757.java:816)
	at net.minecraft.class_310.method_1523(class_310.java:1177)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 1 total; [class_746['yorkmouse'/213, l='ClientLevel', x=-102.39, y=64.50, z=32.55]]
	Chunk stats: 1024, 210
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,87,0), Section: (at 0,7,0 in 0,5,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 467 game time, 467 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:453)
	at net.minecraft.class_310.method_1587(class_310.java:2341)
	at net.minecraft.class_310.method_1514(class_310.java:787)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

**Comment by cScotPlay** [comment]
Thank you for reporting this.  I've tried reproducing this with the current version of 1.18.2 and 1.19.2.  I am unsuccessful in getting this crash report.  Could you provide additional information to allow me to continue trouble shooting?  ---Thanks.

**Comment by yorkmouse** [comment]
I assume it's because I have shaders enabled.
Sodium: 0.4.4
Iris: 1.5.0
Fabric Version: 0.14.13

**Comment by cScotPlay** [comment]
That may be the case but I also run shaders whenever I play.  Haven't yet tested against it but its definitely something I want to fix.  Thanks I'll get after seeing what's going on.

**Comment by redstoneguy10ls** [comment]
I've had this happen to me as well, would be great if it got patched soon as I cant log into my worlds anymore

**Comment by Snownee** [comment]
```
Description: Rendering Block Entity

java.lang.ExceptionInInitializerError
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.render(CrushingTubRenderer.java:40)
	at net.mcs3.rusticated.client.screens.renderer.CrushingTubRenderer.method_3569(CrushingTubRenderer.java:23)
	at net.minecraft.class_824.method_23079(class_824.java:90)
	at net.minecraft.class_824.method_23081(class_824.java:79)
	at net.minecraft.class_824.method_3554(class_824.java:104)
	at net.minecraft.class_824.method_3555(class_824.java:79)
	at me.jellysquid.mods.sodium.client.render.SodiumWorldRenderer.renderTileEntities(SodiumWorldRenderer.java:281)
	at net.minecraft.class_761.handler$elh000$sodium$onRenderTileEntities(class_761.java:13132)
	at net.minecraft.class_761.method_22710(class_761.java:1364)
	at net.minecraft.class_757.method_3188(class_757.java:1039)
	at net.minecraft.class_757.method_3192(class_757.java:816)
	at net.minecraft.class_310.method_1523(class_310.java:1177)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1058.method_4598()" because "sprite" is null
	at com.lowdragmc.shimmer.client.model.ShimmerMetadataSection.isBloom(ShimmerMetadataSection.java:40)
	at net.minecraft.class_777.handler$eim000$shimmer$pTintIndex(class_777.java:519)
	at net.minecraft.class_777.<init>(class_777.java:19)
	at net.fabricmc.fabric.api.renderer.v1.mesh.QuadView.toBakedQuad(QuadView.java:129)
	at net.mcs3.rusticated.util.RenderUtility.<clinit>(RenderUtility.java:55)
	... 18 more
```

**Comment by cScotPlay** [comment]
Thanks for identifying this. The issue lies with the addition of Sodium. Add Indium and this will fix the issue. Future version has a warning screen that will identify this issue.

**Comment by cScotPlay** [comment]
Issue fixed in version 2.0.6
